### PR TITLE
Check for sinol-make updates when running

### PR DESCRIPTION
**Pull request by MasloMaslane**

### CODE REVIEW

**Comment by tonowak** [comment]
How does it work when there is no internet connection or it is very slow? In such cases, we shouldn't annoy the user in any way.

**Comment by MasloMaslane** [comment]
When there’s no internet the code will work well, but I haven’t thought about slow connections

**Comment by MasloMaslane** [comment]
I added a one second timeout for the request

**Comment by tonowak** [comment]
Can this be reduced to e.g. 0.3s? We should be as non-invasive as possible. In fact, if it's possible to cache when we have last checked for updates, it would be even better, instead of executing it on each `run` command.

File: src/sinol_make/util.py [hunk 1]
```diff
@@ -154,6 +157,69 @@ def save_config(config):
...
+    if not data_dir.is_dir():
+        os.mkdir(data_dir)
+
+    thread = threading.Thread(target=check_version)
+    thread.start()
+    version_file = data_dir.joinpath("version")
+    if version_file.is_file():
+        version = version_file.read_text()
+        try:
+            if compare_versions(current_version, version) == -1:
```
Comment (tonowak) [inline src/sinol_make/util.py:176]: What if we just updated the version? This file probably will still exist (the `check_version` won't be fast enough to overwrite here) and the warning will be printed, even though we have just updated the version.

**Review by tonowak — COMMENTED** [review]
Can I ask to add a few more comments? It's not immediately obvious what the new code does, what is being done concurrently, why do we have a new directory, what are possible orders of executions.

**Comment by tonowak** [comment]
Can we add tests? I want to be sure that the warning isn't printed when it shouldn't be printed and the code doesn't crash. So we could e.g. test the code when there is no internet connection, or with different release versions (that would require somehow changing the link or the behavior, so that testing it will be possible).

File: src/sinol_make/util.py (hunk 1 above)
Comment (MasloMaslane) [reply src/sinol_make/util.py:176]: Then the current version will be the same (or higher) than the one in the file, so I don't think there will be a problem

Comment (tonowak) [reply src/sinol_make/util.py:176]: I see, you are right. Nevertheless, adding comments explaining why this method is correct would improve readability and maintainability.

File: src/sinol_make/__init__.py [hunk 2]
```diff
@@ -32,6 +32,11 @@ def main():
 
     for command in commands:
         if command.get_name() == args.command:
+            new_version = util.check_for_updates(__version__)
+            if new_version is not None:
+                print(util.warning(f'New version of sinol-make is available (your version: {__version__}, available version: {new_version}).\n'
+                                   f' You can update it by running `pip install sinol-make --upgrade`.'))
```
Comment (tonowak) [inline src/sinol_make/__init__.py:38]:
```suggestion
                                   f' You can update it by running `pip3 install sinol-make --upgrade`.'))
```

**Review by tonowak — APPROVED** [review]

**Pull request merged**